csrc/rope.cu

```diff
@@ -259,3 +259,98 @@ void apply_llama31_rope_pos_ids(at::Tensor q, at::Tensor k, at::Tensor q_rope, a
     });
   });
 }
+
+void mla_rope_quantize(at::Tensor q_rope_in, at::Tensor k_rope_in, at::Tensor q_nope_in,
+                       at::Tensor k_nope_in, at::Tensor q_rope_out, at::Tensor k_rope_out,
+                       at::Tensor q_nope_out, at::Tensor k_nope_out, at::Tensor cos_sin_cache,
```

Comment:
Can we also fuse q_nope * W_KC bmm in this kernel too? 

In [SGlang](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/models/deepseek_v2.py#L1226-L1256) for example, the query is divided to `q_rope` and `q_nope` -> bmm between `q_nope` and `w_kc` and then rope on `q_rope` and `k_rope`.